### ISSUE TITLE
Jupyter kernel crashes on an R's Error: attempt to use zero-length variable name

### ISSUE THREAD

**Issue by lgautier** [rpy2/rpy2-migrate-issues#601]
**[Original report](https://bitbucket.org/rpy2/rpy2/issue/601) by Michał Krassowski (Bitbucket: [krassowski](https://bitbucket.org/krassowski), GitHub: [krassowski](https://github.com/krassowski)).**

----------------------------------------

Assigning a list with zero-length variable name is an error, but it is well-tolerated in R interactive sessions \(the user gets `Error: attempt to use zero-length variable name` message and the session continues normally\). However, the Jupyter kernel crashes in the interactive rpy2 use when this line is run:

```r
%R list(''='zero-length variable name')
```

Following communicates are written out:

```
R[write to console]: Error: attempt to use zero-length variable name
R[write to console]: Execution halted
```

And a kernel crash pop-up follows:  `<Kernel Restarting> The kernel for notebook.ipynb appears to have died. It will restart automatically.` A loosely-related issue was #409.

  
For me, this is only a minor nuisance, but it would be nice if - similarly to R interactive sessions - this error did not result in a crash of the kernel \(as this can lead to a waste of minutes and sometimes hours of computations\).

‌

rpy2 3.2.2; JupyterLab 1.2.3; Python 3.7.5; R 3.6.1

**Comment by lgautier** [comment]
@krassowski : copied from bitbucket (let me know if you'd like to make your own copy instead so you can edit it).

> I admire your work on rpy2, tracking this down to a memory leak in R must have been interesting.
>
> I got another syntax error which leads to:
> 
> ```
> R[write to console]: Fatal error: unable to initialize the JIT
> R[write to console]: Execution halted
> ```
>
> And this is, again, a slight annoyance.
> 
> User story: in R one can convert latex to plotmath using latex2exp package. However, we need to  escape the backslash character to prevent it from being interpreted as a special character sequence (like a new line or caret return). Failing to escape will lead to an error, but again should not break the interactive session. But it does (kernel dies).
> 
> An example:
> 
> ```
> >>> TeX(paste0("Pearson's Chi-squared test: \chi^2=", chisq$statistic))
> R[write to console]: Error: '\c' is an unrecognized escape in character string starting ""Pearson's Chi-squared test: \c"
> R[write to console]: Fatal error: unable to initialize the JIT
> R[write to console]: Execution halted
> ```
> Again, this can be frustrating to lose data, especially if someone instinctively writes “normal” (unescaped) latex.
>
> Expected behaviour: I get an error message, fix the typo and carry on.
>
> All the versions remain unchanged (as in my first post).
>
> Edit: please, let me know if you would like this in a new issue, I can fill in the new issue form - just though that his may have a common origin.

**Comment by lgautier** [comment]
I asked about the issue on the r-devel mailing list. One response does not think that the memory leak mentioned in the code comment is related.

The other segfault you mentioned is indeed there and easy to trigger.
```
>>> import rpy2.rinterface as ri
>>> ri.initr()
>>> ri.parse('x+"\c"')
R[write to console]: Fatal error: unable to initialize the JIT


*** stack smashing detected ***: <unknown> terminated
Aborted (core dumped)
```

I am still trying to figure out what might be going on here (weird `jump` in on the R side landing in the wrong place because this is embedded in Python ? Issue with balancing `PROTECT`/`UNPROTECT` ?

**Comment by lgautier** [comment]
No segfault with the following change:

```
# ri.parse('x+"\c"')
ri.baseenv['parse']('x+"\c"')
```

I am increasingly thinking that `R_ParseVector()` in the C API is not well-behaved / compatible with its use in an embedded R...

The performance penalty of moving away from it and use the R function is likely negligible comparable to the expected gain in stability, and this would allow me to quickly provide a fix for this.

**Comment by lgautier** [comment]
Fixed in master: https://github.com/rpy2/rpy2/commit/c62e57a4c7e21fcfe4cb129bf36155f7599f33e4

**Comment by lgautier** [comment]
The fix was backported to the 3.2.x series. It is available in version 3.2.3, just released and available on pypi.
